content/deployment.md

```diff
@@ -61,11 +61,39 @@ It's always a good idea to use SSL for Meteor applications (see the [Security Ar
+The basic way to use a CDN is to upload your files to the CDN and change your URLs to point at the CDN (for instance if your Meteor app is at `http://myapp.com`, changing your image URL from `<img src="http://myapp.com/cats.gif">` to `<img src="http://mycdn.com/cats.gif">`). However, this would be hard to do with Meteor, since the largest file – your Javascript bundle – changes every time you edit your app.
 
-If you are following the above approach, you may also want to manually add the CDN's hostname whenever you put an image/other asset URL in your application's code. To do this throughout your app, you can write a generic helper like `imageUrl()`.
+For Meteor, we recommend using a CDN with "origin" support (like [CloudFront](http://joshowens.me/using-a-cdn-with-your-production-meteor-app/)), which means that instead of uploading your files in advance, the CDN automatically fetches them from your server. You put your files in `public/` (in this case `public/cats.gif`), and when your Australian user asks the CDN for `http://mycdn.com/cats.gif`, the CDN, behind the scenes, fetches `http://myapp.com/cats.gif` and then delivers it to the user. While this is slightly slower than getting `http://myapp.com/cats.gif` directly, it only happens one time, because the CDN saves the file, and all subsequent Australians who ask for the file get it quickly.
+
+To get Meteor to use the CDN for your Javascript and CSS bundles, call `WebAppInternals.setBundledJsCssPrefix("mycdn.com")` on the server. This will also take care of relative image URLs inside your CSS files. If you need to use a dynamic prefix, you can return the prefix from a function passed to `WebAppInternals.setBundledJsCssUrlRewriteHook()`.
+
+For all your files in `public/`, change their URLs to point at the CDN. You can use a helper like `assetUrl`.
+
+Before:
+
+```html
+<img src="http://myapp.com/cats.gif">
+```  
+
+After:
+
+```js
+Template.registerHelper("assetUrl", (asset) => {
+  return "http://mycdn.com/" + asset
+});
+```
+
+```html
+<img src="{{assetUrl "cats.gif"}}">
+```
+
+<h4 id="cordova-assets">Cordova assets</h4>
+
+In Meteor versions 1.3 and up, if your app is mainly meant to be used inside Cordova, we recommend **not** using a CDN. Meteor Cordova caches your `myapp.com/*` files on the phone, and it's much faster to get the phone copies than request them from a CDN.
+
+Pre-1.3 Meteor Cordova still cached your files on the phone, but it re-downloaded *all* the files during each hot code push, and it did so from main-thread Javascript, which froze the UI. In order to shorten the duration of the UI freeze, you had to remove the files from `public/` and upload them to a CDN. 
```

Comment:
I'd leave this out, because the mobile guide is only meant for 1.3. Also, I would advise anyone still on a pre-1.3 version of Meteor to upgrade rather than try to work around these issues by uploading all your assets to a CDN (which has other downsides, because now the assets are no longer available locally).